### DISCUSSION TITLE
Writing to /tmp folder, limits ?

### DISCUSSION THREAD

**Discussion by ErnestBrandi** [vercel/vercel#5190] [Help] [answered]
Hi, I can't find any real info about writing files on Vercel instances. 

What I know is (if what I read is correct) : 
- you can write files only in the  /tmp folder
- only the same process can handle that file
- you can't rely on the way and time that files are stored

What I would like to know : 
- is this a bad habit to write stuffs on disk when hosting a website on Vercel ?
- what is the limit (file size) I can write to the /tmp folder ? 
- where can I find infos about this ? (aws ?)

What I wanna do : retrieving documents (zip files) from an api, extract them and serving them to the client.

Currently I am using [yauzl](https://www.npmjs.com/package/yauzl) to extract data directly from a `Buffer`, then I `pipe` the extracted file to the client (I'm using Node.js). But on big files I am facing the current issue `LAMBDA_RUNTIME Failed to post handler success response. Http response code: 413`. So I though I could simply write in that /tmp folder in order to do the same process. I that achievable ? If not, could someone point me to the right direction ?
Thanks

**Comment by ErnestBrandi** [comment, marked as answer]
Ended up storing files on AWS.

**Comment by paulogdm** [comment]
> is this a bad habit to write stuffs on disk when hosting a website on Vercel ?

Not at all! For temporary storage, it should be fine. If you need persistence, however, a DB or bucket is recommended.

> what is the limit (file size) I can write to the /tmp folder ?

I can check that information. However, I don't think it is meaningful since heavy-files should use specific solutions like buckets.

> What I wanna do : retrieving documents (zip files) from an api, extract them and serving them to the client.

You need to use AWS S3 or a similar solution for this.

**Reply by ErnestBrandi** [reply]
Thanks for you reply. 

> You need to use AWS S3 or a similar solution for this.

Indeed and this is what I do now. Because even if you can write stuffs in `tmp` folder, then you can't access them from the browser (only files under `public` folder can be).

**Reply by amanbthakkar** [reply]
@ErnestBrandi hi! I am facing the same problem as you and looking at this I have come to the conclusion that storing files on /tmp isn't a viable solution. I need to store some light but important data in files which I need across the sleep sessions but I've never worked with AWS yet. Can you help me with a guide to how you used AWS for this? Any links I can look at? TIA!

**Reply by ErnestBrandi** [reply]
Hi, just check S3 doc depending on the language you are using : https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/s3-example-creating-buckets.html. You can still write to `tmp` folder then use `s3.upload` to write that file on your bucket (= aws disk space). Also if your data is concise you can use something as https://lambda.store/ instead; and if your data is not highly secure you can encrypt it and store it client side with cookies.
